docs/user-guide/relay-nodes.md

```diff
@@ -23,3 +40,70 @@ If you're using the Nexodus rpm package, edit `/etc/sysconfig/nexodus` to add th
+
+### Using public DNS name
+
+User need to configure a DNS name (e.g NameCheap, CloudFlare etc) resolving to the IP address of the machine running the relay node. The DNS name will be used to generate the TLS certificate (letsencrypt) for the relay node. User can use the following command to onboard the relay node:
+
+```sh
+sudo nexd relayderp --hostname <relay domain name> --onboard
+```
+
+### Using private DNS name
+
+If user don't want to use a public DNS name, user can use a private DNS name (e.g. `xyz.relay.io`) to onboard the relay. But this requires users to generate a TLS certificate and key for the relay node. User can use the following commands to generate the TLS certificate and key (you need openssl to generate the key and certificate):
+
+```sh
+# Create CA’s root key
+openssl genrsa -out rootCA.key 2048
+
+# Create CA’s root certificate, that will be used to sign the server certificate
+openssl req -x509 -new -nodes -key rootCA.key -sha256 -days 1024 -out rootCA.pem
+
+# Create server key and certificate sign request
+openssl req -new -newkey rsa:2048 -sha256 -nodes -keyout server.key -out server.csr
+```
+
+Create an extension file
+
+```sh
+# cat v3.ext
+authorityKeyIdentifier=keyid,issuer
+basicConstraints=CA:FALSE
+keyUsage = digitalSignature, nonRepudiation, keyEncipherment, dataEncipherment
+subjectAltName = <relay-private-domain-name>
+```
+
+Set subjectAltName to the private DNS name (e.g. `xyz.relay.io`) that user want to use for the relay node.
+
+Run the following command to sign and generate the server certificate
+
+```sh
+# Sign and create server certificate , do not use more than 825 days for certificate, otherwise Darwin OS will throw error
+openssl x509 -req -in server.csr -CA rootCA.pem -CAkey rootCA.key -CAcreateserial -out server.crt -days 800 -sha256 -extfile v3.ext
+```
+
+Above command will generate rootCA key (rootCA.key) and certificate (rootCA.cet) files and server key (server.key) and certificate (server.crt) files. Rename the server key and certificate files to `<relay-private-domain-name>.key` and `<relay-private-domain-name>.crt` respectively.
+
+```sh
+cp server.key xyz.relay.io.key
+cp server.crt xyz.relay.io.crt
+```
+
+Now you can use the private DNS name and the key/certificates to onboard the relay. Use the following command to onboard the relay node:
+
+```sh
+sudo nexd relayderp --hostname xyz.relay.io -a ":443" --certmode manual --certdir "<certificate-directory-path>"  --onboard
+```
+
+In this scenario user need to install the root CA key (rootCA.key) to all the nodes that needs to be connected through this onboard relay. Copy the file to each node and run the following command to install the root CA key:
+
+```sh
+CAROOT=</path/to/cert/dir> mkcert -install
+```
+
+Now you can onboard the device to Nexodus and if the device is behind symmetric NAT, it will use the self hosted relay and will use the root CA key to verify the TLS certificate of the relay node.
```

Comment:
```suggestion
Now you can onboard the device to Nexodus, and if the device is behind symmetric NAT, it will use the self-hosted relay and will use the root CA key to verify the TLS certificate of the relay node.
```